Executor: 21 miscellaneous execution failures

## Problem

21 SQLLogicTest files are failing with various execution errors not related to query result mismatches.

## Error Categories

### 1. Type Mismatch Errors (3 failures)
```
✗ index/random/10/slt_good_1.test
Error: TypeMismatch { left: Varchar("mguub"), op: "unary +", right: Null }

Example queries attempting unary + operator on VARCHAR columns
```

**Root Cause**: The unary + operator implementation in `crates/vibesql-executor/src/evaluator/expressions/operators.rs:13-89` only handles numeric types (Integer, Smallint, Bigint, Float, Real, Double, Numeric) and NULL. Non-numeric types like VARCHAR are rejected with a TypeMismatch error.

**Implementation Location**: `eval_unary_op()` function at `crates/vibesql-executor/src/evaluator/expressions/operators.rs:13-89`

**Expected Behavior**: According to SQL standard behavior, unary + on non-numeric types should either:
- Return a type error (current behavior, which may be correct)
- OR attempt implicit casting to numeric type first (MySQL behavior)

Need to investigate what SQLLogicTest expects in these cases.

### 2. View Dependency Errors (1 failure)
```
✗ index/view/10/slt_good_1.test  
Error: Other("View or table 'VIEW_1_TAB0_153' is still in use by 1 view(s): VIEW_3_TAB0_153")

Issue: DROP VIEW failing when dependent views exist
```

**Root Cause**: The view dependency tracking in `crates/vibesql-catalog/src/store/advanced/views.rs:34-74` correctly identifies dependent views and returns `ViewInUse` error when CASCADE is not specified. However, the test is likely expecting the view to be dropped with CASCADE behavior, or the test setup is incorrect.

**Implementation Location**: 
- `drop_view()` at `crates/vibesql-catalog/src/store/advanced/views.rs:34-74`
- `find_dependent_views()` at `crates/vibesql-catalog/src/store/advanced/views.rs:76-93`
- Error conversion at `crates/vibesql-executor/src/errors.rs:682-686`

**Possible Solutions**:
1. The test may be using `DROP VIEW` without CASCADE when it should use `DROP VIEW ... CASCADE`
2. The dependency tracking may have a bug (e.g., case sensitivity issues in view name matching)
3. Need to verify the expected behavior in the original SQLite test case

### 3. Column Resolution Errors (1 failure)
```
✗ index/view/10/slt_good_2.test
Error: ColumnNotFound { column_name: "PK", table_name: "unknown", searched_tables: ["TAB0_302"], available_columns: ["column1", "column2"] }

Issue: Column name resolution failing in views
```

**Root Cause**: Column resolution is failing when querying from views. This suggests the view's column metadata or alias handling may not be preserved correctly when the view is materialized.

**Related Code Paths**:
- View execution and column resolution
- Table/view metadata retrieval
- Column name normalization (case sensitivity)

**Possible Issues**:
1. View column aliases not being stored/retrieved correctly
2. Case sensitivity mismatch between view definition and query
3. View's underlying table column mapping broken

### 4. Other Execution Errors (~16 failures)
Various other execution issues including:
- Expression evaluation errors
- Schema resolution issues  
- Constraint validation problems

Need to run individual test files to categorize these errors properly.

## Impact

- 21/623 tests failing (3.4% of test suite)
- Affects view functionality, type coercion, and expression evaluation
- Lower priority than the 555 query result mismatches (#1634)
- Higher priority than the 16 parser errors (#1635)

## Investigation Steps

### Step 1: Categorize Remaining Errors
Run the failing tests individually to categorize the ~16 uncategorized failures:
```bash
# Get list of all 21 failing tests
./scripts/sqllogictest test <each-failing-test> --verbose
```

### Step 2: Fix Type Mismatch Errors
1. **Analyze expected behavior**: Check what SQLite/MySQL do with `+VARCHAR_COLUMN`
2. **Decision point**: Should we:
   - Keep the error (strict type checking)
   - Add implicit string-to-number conversion
   - Handle VARCHAR specially (return NULL or convert to 0)
3. **Update `eval_unary_op()`** in `crates/vibesql-executor/src/evaluator/expressions/operators.rs:13-89`

### Step 3: Fix View Dependency Errors
1. **Run the failing test** with verbose output to see exact SQL commands
2. **Check if CASCADE is needed**: Does the test expect `DROP VIEW ... CASCADE`?
3. **Debug dependency tracking**: Add logging to `find_dependent_views()` to see what views it's finding
4. **Verify case sensitivity**: Ensure view name matching respects the case_sensitive_identifiers setting

### Step 4: Fix Column Resolution Errors
1. **Investigate view column metadata storage**: Check how columns are stored when view is created
2. **Trace column resolution**: Add logging during SELECT from view to see where "PK" lookup fails
3. **Check alias handling**: Verify that column aliases in view definitions are preserved
4. **Test case sensitivity**: The error shows "unknown" table which suggests a metadata lookup issue

### Step 5: Address Remaining Errors
After categorizing the ~16 other errors, create focused subtasks for each category.

## Acceptance Criteria

- [ ] All 21 execution errors are categorized and documented
- [ ] Type mismatch errors (3 tests) are fixed with correct SQL semantics
- [ ] View dependency errors (1 test) are fixed 
- [ ] Column resolution errors (1 test) are fixed
- [ ] Remaining execution errors (~16 tests) have fixes or documented decisions
- [ ] All 21 tests pass
- [ ] No regressions in existing passing tests

## Test Commands

```bash
# Type mismatch example
./scripts/sqllogictest test third_party/sqllogictest/test/index/random/10/slt_good_1.test --verbose

# View dependency example  
./scripts/sqllogictest test third_party/sqllogictest/test/index/view/10/slt_good_1.test --verbose

# Column resolution example
./scripts/sqllogictest test third_party/sqllogictest/test/index/view/10/slt_good_2.test --verbose
```

## Related Issues

- #1634 - 555 query result mismatches (higher priority)
- #1635 - 16 parser errors
- #1606 - Overall SQLLogicTest progress tracking

## Technical References

**Key Files**:
- `crates/vibesql-executor/src/evaluator/expressions/operators.rs` - Unary operator evaluation
- `crates/vibesql-executor/src/evaluator/operators/arithmetic/mod.rs` - Arithmetic type coercion
- `crates/vibesql-catalog/src/store/advanced/views.rs` - View dependency tracking
- `crates/vibesql-executor/src/errors.rs` - Error type definitions
- `crates/vibesql-catalog/src/errors.rs` - Catalog error types

**Testing Infrastructure**:
- `./scripts/sqllogictest` - Test runner script
- `third_party/sqllogictest/test/` - Test file location

## Progress Update

While investigating this issue, discovered and fixed a **compilation blocker** that was preventing the sqllogictest suite from running.

### Fixed: Compilation Error

**PR #1659** - Fixes sqllogictest compilation error introduced by PR #1639

The sqllogictest adapter was missing handling for the  statement, causing compilation to fail with:
```
error[E0004]: non-exhaustive patterns: vibesql_ast::Statement::SetVariable(_) not covered
```

This was blocking all sqllogictest runs and preventing investigation of the execution failures tracked in this issue.

### Current Status  

- ✅ **MERGED** (PR #1637): Type mismatch errors for unary + on text types (3 failures fixed)
- ✅ **CREATED** (PR #1659): Compilation blocker fix  
- ⏳ **TODO**: Full sqllogictest run is currently executing to categorize remaining execution failures
- ⏳ **TODO**: Investigate and fix view dependency errors (1 failure)
- ⏳ **TODO**: Investigate and fix column resolution errors (1 failure)
- ⏳ **TODO**: Categorize and address other execution errors (~16 failures)

### Next Steps

Once PR #1659 is merged and the full test run completes, I'll categorize the remaining execution failures and create targeted fixes for each category.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/vibesql-executor/src/select/executor/execute.rs
+++ b/crates/vibesql-executor/src/select/executor/execute.rs
@@ -109,7 +109,7 @@ impl SelectExecutor<'_> {
         cte_results: &HashMap<String, CteResult>,
     ) -> Result<FromResult, ExecutorError> {
         use crate::select::scan::execute_from_clause;
-        execute_from_clause(from, cte_results, self.database, None, |query| self.execute(query))
+        execute_from_clause(from, cte_results, self.database, None, |query| self.execute_with_columns(query))
     }
 
     /// Execute a FROM clause with WHERE clause for predicate pushdown
@@ -121,7 +121,7 @@ impl SelectExecutor<'_> {
     ) -> Result<FromResult, ExecutorError> {
         use crate::select::scan::execute_from_clause;
         execute_from_clause(from, cte_results, self.database, where_clause, |query| {
-            self.execute(query)
+            self.execute_with_columns(query)
         })
     }
 }

--- a/crates/vibesql-executor/src/select/scan/derived.rs
+++ b/crates/vibesql-executor/src/select/scan/derived.rs
@@ -3,7 +3,7 @@
 //! Handles execution of subqueries in FROM clauses (derived tables)
 //! by executing the subquery and wrapping it with an alias.
 
-use crate::{errors::ExecutorError, schema::CombinedSchema};
+use crate::{errors::ExecutorError, schema::CombinedSchema, select::SelectResult};
 
 /// Derive a column name from an expression (simplified version from columns.rs)
 fn derive_column_name_from_expr(expr: &vibesql_ast::Expression) -> String {
@@ -87,10 +87,12 @@ pub(crate) fn execute_derived_table<F>(
     execute_subquery: F,
 ) -> Result<super::FromResult, ExecutorError>
 where
-    F: Fn(&vibesql_ast::SelectStmt) -> Result<Vec<vibesql_storage::Row>, ExecutorError>,
+    F: Fn(&vibesql_ast::SelectStmt) -> Result<SelectResult, ExecutorError>,
 {
-    // Execute subquery to get rows
-    let rows = execute_subquery(query)?;
+    // Execute subquery to get rows and column names
+    let subquery_result = execute_subquery(query)?;
+    let rows = subquery_result.rows;
+    let subquery_columns = subquery_result.columns;
 
     // Derive schema from SELECT list
     let mut column_names = Vec::new();
@@ -100,17 +102,23 @@ where
     for item in &query.select_list {
         match item {
             vibesql_ast::SelectItem::Wildcard { .. } | vibesql_ast::SelectItem::QualifiedWildcard { .. } => {
-                // For SELECT * or SELECT table.*, expand to all columns from the result rows
-                // Since we executed the subquery, the rows tell us how many columns there are
+                // For SELECT * or SELECT table.*, use the column names from the subquery result
+                // This preserves the actual column names instead of generating generic ones
                 if let Some(first_row) = rows.first() {
                     for (j, value) in first_row.values.iter().enumerate() {
-                        let col_name = format!("column{}", col_index + j + 1);
+                        // Use actual column name from subquery if available
+                        let col_name = subquery_columns.get(col_index + j).cloned()
+                            .unwrap_or_else(|| format!("column{}", col_index + j + 1));
                         column_names.push(col_name);
                         column_types.push(value.get_type());
                     }
                     col_index += first_row.values.len();
                 } else {
-                    // No rows, no columns from wildcard
+                    // No rows - use column names from subquery metadata
+                    for col_name in &subquery_columns {
+                        column_names.push(col_name.clone());
+                        column_types.push(vibesql_types::DataType::Null);
+                    }
                 }
             }
             vibesql_ast::SelectItem::Expression { expr, alias: col_alias } => {

--- a/crates/vibesql-executor/src/select/scan/join_scan.rs
+++ b/crates/vibesql-executor/src/select/scan/join_scan.rs
@@ -25,7 +25,7 @@ pub(crate) fn execute_join<F>(
     execute_subquery: F,
 ) -> Result<super::FromResult, ExecutorError>
 where
-    F: Fn(&vibesql_ast::SelectStmt) -> Result<Vec<vibesql_storage::Row>, ExecutorError> + Copy,
+    F: Fn(&vibesql_ast::SelectStmt) -> Result<crate::select::SelectResult, ExecutorError> + Copy,
 {
     // Execute left and right sides with WHERE clause for predicate pushdown
     let left_result = super::execute_from_clause(left, cte_results, database, where_clause, execute_subquery)?;

--- a/crates/vibesql-executor/src/select/scan/mod.rs
+++ b/crates/vibesql-executor/src/select/scan/mod.rs
@@ -45,7 +45,7 @@ pub(super) fn execute_from_clause<F>(
     execute_subquery: F,
 ) -> Result<FromResult, ExecutorError>
 where
-    F: Fn(&vibesql_ast::SelectStmt) -> Result<Vec<vibesql_storage::Row>, ExecutorError> + Copy,
+    F: Fn(&vibesql_ast::SelectStmt) -> Result<super::SelectResult, ExecutorError> + Copy,
 {
     // Check if this is a multi-table join that could benefit from reordering
     if matches!(from, vibesql_ast::FromClause::Join { .. }) {

--- a/crates/vibesql-executor/src/select/scan/reorder.rs
+++ b/crates/vibesql-executor/src/select/scan/reorder.rs
@@ -15,6 +15,7 @@ use crate::{
     select::{
         cte::CteResult,
         join::{nested_loop_join, JoinOrderAnalyzer, JoinOrderSearch},
+        SelectResult,
     },
 };
 
@@ -130,7 +131,7 @@ pub(crate) fn execute_with_join_reordering<F>(
     execute_subquery: F,
 ) -> Result<super::FromResult, ExecutorError>
 where
-    F: Fn(&vibesql_ast::SelectStmt) -> Result<Vec<vibesql_storage::Row>, ExecutorError> + Copy,
+    F: Fn(&vibesql_ast::SelectStmt) -> Result<SelectResult, ExecutorError> + Copy,
 {
     // Step 1: Flatten join tree to extract all tables
     let mut table_refs = Vec::new();

